keepercommander/cli.py

```diff
@@ -70,7 +78,86 @@ def display_command_help(show_enterprise = False, show_shell = False):
     print('Type \'command -h\' to display help on command')
 
 
+msp_params = None
+mc_params_dict = {}
+current_mc_id = None
+
+
+def is_executing_as_msp_admin():
+    return msp_params is not None
+
+
+def check_if_running_as_mc(params, args):
+    has_mc_id_regex = "--mc?\\s\\d+"
+
+    global msp_params
+
+    if "--mc" in args:                                   # Impersonating as Managed Company (MC)
+
+        mc_id = -9
+        try:
+            mc_id = re.search(has_mc_id_regex, args).group(0)  # get id of the MC from args
+            mc_id = int(mc_id.split()[1])                      # get just the ID
+        except AttributeError:
+            logging.error("No Managed company provided")  # apply your error handling
+            raise
+
+        cur_msp_params = params if msp_params is None else params
+
+        query_enterprise(cur_msp_params)
+        query_msp(cur_msp_params)
+
```

Comment:
"query_enterprise" may take awhile to complete. 
Is it really needed in this context?
We are switching to MC and we are going to load MC data anyway.
We need to reload enterprise data when we back to MSP, though.

Reply:
This was done to make sure data is refreshed even when switching. Removed these lines.
Also added reload of the data when switching back to MSP